src/ModelStore.ts

```diff
@@ -92,7 +92,14 @@ export class ModelStore {
             if (data && data[Constants.ITEMS_PROP]) {
                 const localData = clone(newData);
                 const items = data[Constants.ITEMS_PROP] || {};
-
+                const oldData = items[itemKey];
```

Comment:
The solution seems good.
Please increase code coverage
One thing comes to my mind, what happens if the component is saved and has text:undefined in it's state (i.e we modify a different parameter and save is invoked)? Is it propagated to backend how does backend react to that?
Nice that we have Angular and React covered in one place!

Reply:
Empty text is not propagated to backend
Since we are using current data as reference to check with new changes there will never be a condition where text will be undefined as current data wont have keys for undefined properties